### PR TITLE
[Chore] Setup eslint for typescript assets

### PR DESCRIPTION
**Pull request by jmrossy**
- Setup shared eslint config for whole monorepo
- Create lint-ts job for linting typescript files
- Run lint-ts in CI

### CODE REVIEW

File: typescript/infra/src/agents/aws/key.ts [hunk 1]
```diff
@@ -92,9 +92,11 @@ export class AgentAwsKey extends AgentKey {
   }
 
   async createIfNotExists() {
-    let keyId = await this.getId();
+    const keyId = await this.getId();
     // If it doesn't exist, create it
     if (!keyId) {
+      // TODO should this be awaited? create is async
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
```
Comment (nambrot) [inline typescript/infra/src/agents/aws/key.ts:99]: Yes I think this should be

File: typescript/infra/src/agents/aws/key.ts [hunk 2]
```diff
@@ -208,6 +210,8 @@ export class AgentAwsKey extends AgentKey {
     await client.send(new DeleteAliasCommand({ AliasName: newAlias }));
 
     // Address should have changed now
+    // TODO should this be awaited? fetch is async
```
Comment (nambrot) [inline typescript/infra/src/agents/aws/key.ts:213]: Agreed, or returned

File: typescript/infra/src/agents/gcp.ts [hunk 3]
```diff
@@ -114,6 +114,7 @@ export class AgentGCPKey extends AgentKey {
     }
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (nambrot) [inline typescript/infra/src/agents/gcp.ts:117]: Why is this unused?

File: typescript/sdk/src/app.ts [hunk 4]
```diff
@@ -38,7 +38,7 @@ export class AbacusApp<
     return this.get(chain).addresses;
   }
 
-  reconnect(chain: Chain, connection: Connection) {
+  reconnect(chain: Chain, connection: Connection): void {
```
Comment (nambrot) [inline typescript/sdk/src/app.ts:41]: Do method signatures now always have to be explicit?

**Review by nambrot — APPROVED** [review]
Thank you 🙏

Only wondering whether requiring explicit method signatures is a bit heavy

**Comment by jmrossy** [comment]
> Thank you 🙏
> 
> Only wondering whether requiring explicit method signatures is a bit heavy

It's only enabled in the SDK and only because that's the setting that was already there (albeit unused). It can be nice for an SDK as it may improve the auto-generated docs but not strictly necessary. Lmk if you'd prefer it disabled for now

File: typescript/sdk/src/app.ts (hunk 4 above)
Comment (jmrossy) [reply typescript/sdk/src/app.ts:41]: Only in the SDK package

File: typescript/infra/src/agents/gcp.ts (hunk 3 above)
Comment (jmrossy) [reply typescript/infra/src/agents/gcp.ts:117]: Dunno!

**Pull request merged**